.github/workflows/codeql.yml

```diff
@@ -0,0 +1,48 @@
+name: "CodeQL PR Analysis"
+
+on:
+    workflow_call:
+    workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Analyze (${{matrix.language}})
+    runs-on: 'ubuntu-latest'
```

Comment:
If this check takes longer than 2 minutes or so, we should run it on-prem

Reply:
I agree, see slack channel. I just can't test on on-prem runners and don't want to PR something I am not sure is working.